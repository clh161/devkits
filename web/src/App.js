// @flow strict

import React from 'react';
import Dashboard from './component/Dashboard';
import HTMLEncoder from './page/HTMLEncoder';
import type { Node } from 'react';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import TimestampConverter from './page/TimestampConverter';
import Home from './page/Home';
import CaseTypeConverter from './page/CaseTypeConverter';

export default function App(): Node {
  return (
    <Router>
      <div>
        <Switch>
          <Dashboard>
            <Route exact path="/">
              <Home />
            </Route>
            <Route path="/html-encoding">
              <HTMLEncoder />
            </Route>
            <Route path="/unix-timestamp-converter">
              <TimestampConverter />
            </Route>
            <Route path="/case-type-converter">
              <CaseTypeConverter />
            </Route>
          </Dashboard>
        </Switch>
      </div>
    </Router>
  );
}
