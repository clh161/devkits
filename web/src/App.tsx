import { ThemeProvider } from '@emotion/react';
import React, { ReactElement } from 'react';

import Router from './component/Router';
import DKTheme from './DKTheme';

const App = (): ReactElement => (
  <ThemeProvider theme={DKTheme}>
    <Router />
  </ThemeProvider>
);

export default App;
