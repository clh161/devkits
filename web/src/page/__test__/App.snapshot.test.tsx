import React from 'react';
import { Helmet } from 'react-helmet';
import renderer from 'react-test-renderer';

import App from '../../App';
it('Init with default', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(Helmet.peek()).toMatchSnapshot();
});
