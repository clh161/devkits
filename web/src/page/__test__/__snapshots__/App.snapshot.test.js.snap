// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init with default 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <h1
              className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                href="/"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Devkits
              </a>
            </h1>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-end MuiGrid-grid-xs-1"
          >
            <a
              aria-disabled={false}
              aria-label="Github Repository"
              className="MuiButtonBase-root MuiIconButton-root"
              href="https://github.com/clh161/devkits"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              target="_blank"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <nav
        aria-label="main mailbox folders"
        className="MuiList-root MuiList-padding"
      >
        <a
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          href="/html-encoding"
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              HTML Encoder
            </span>
          </div>
        </a>
        <a
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          href="/url-encoding"
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4.5 11h-2V9H1v6h1.5v-2.5h2V15H6V9H4.5v2zm2.5-.5h1.5V15H10v-4.5h1.5V9H7v1.5zm5.5 0H14V15h1.5v-4.5H17V9h-4.5v1.5zm9-1.5H18v6h1.5v-2h2c.8 0 1.5-.7 1.5-1.5v-1c0-.8-.7-1.5-1.5-1.5zm0 2.5h-2v-1h2v1z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              URL Encoder
            </span>
          </div>
        </a>
        <a
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          href="/unix-timestamp-converter"
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
              <path
                d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Unix Timestamp Converter
            </span>
          </div>
        </a>
        <a
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          href="/case-type-converter"
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.5 4v3h5v12h3V7h5V4h-13zm19 5h-9v3h3v7h3v-7h3V9z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Case Type Converter
            </span>
          </div>
        </a>
        <a
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          href="/csv-editor"
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 9h4V5H3v4zm0 5h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zM8 9h4V5H8v4zm5-4v4h4V5h-4zm5 9h4v-4h-4v4zM3 19h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zm5 0h4v-4h-4v4zm0-14v4h4V5h-4z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              CSV Editor
            </span>
          </div>
        </a>
      </nav>
    </div>
  </div>
  <main
    className="makeStyles-content-3"
  >
    <div
      className="MuiContainer-root makeStyles-container-4 MuiContainer-maxWidthLg"
    >
      <div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <h1
              block={true}
              className="MuiTypography-root MuiTypography-h4"
            >
              Devkits
            </h1>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              An opensource project of collection of development tools
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <span
              className="MuiTypography-root MuiTypography-overline"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://github.com/clh161/devkits"
                onBlur={[Function]}
                onFocus={[Function]}
                target="_blank"
              >
                Github Repository
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Init with default 2`] = `
Object {
  "baseTag": Array [],
  "bodyAttributes": Object {},
  "defer": true,
  "encode": true,
  "htmlAttributes": Object {},
  "linkTags": Array [],
  "metaTags": Array [
    Object {
      "content": "Collection of development tools",
      "name": "description",
    },
    Object {
      "content": "web, web development, development, development tools, devkits",
      "name": "keywords",
    },
  ],
  "noscriptTags": Array [],
  "onChangeClientState": [Function],
  "scriptTags": Array [],
  "styleTags": Array [],
  "title": "Devkits",
  "titleAttributes": Object {},
}
`;
